### DISCUSSION TITLE
Can't use Suspense for useQueries?

### DISCUSSION THREAD

**Discussion by parkgang** [TanStack/query#3422] [Q&A] [answered]
Hi, I would like to run multiple queries in parallel while using Suspense.

The methods of parallel execution may include how to declare useQuery side by side and how to use use useQueries.

In Suspense mode, useQueries was used for parallel query, but Suspense was not executed and it behaved as if Suspense mode was disabled.

But what I want is that `<Suspense />` runs until the parallel call ends, and when `<Suspense />` is removed, I hope that all the queries executed in parallel have been completed.

Is this impossible? As a comparative example, recoil confirmed that I could reach my desired goal through [waitForAll](https://recoiljs.org/docs/api-reference/utils/waitForAll/) function in Suspense mode. Thanks to these examples, I learned that it can be implemented in Suspense mode.

I watched #1523 and waited for it to be fixed, but when I followed issue, I saw that API Spec was changed in v4, so I tested it, and it was the same.

Accordingly, please answer whether I am presenting an impossible function or am planning to modify it.

I have prepared a [sample code](https://codesandbox.io/s/suspense-parallel-query-6d40mm) for clear reproduction.

**Thank you.**

**Comment by TkDodo** [comment]
it's like one of the three or so open bugs we have:
- https://github.com/tannerlinsley/react-query/issues/1523

There have been various attempts at making it work, but it's not trivial, as suspense stops execution when the first promise is thrown.

Have a look at the discussion here as well:
- https://github.com/tannerlinsley/react-query/issues/2923#issuecomment-966703628

**Comment by SunStupic** [comment, marked as answer]
This is [supported](https://github.com/TanStack/query/pull/4498) now. The above comment should not be marked as answer.
